tests/CloneTest.php

```diff
@@ -45,6 +45,19 @@ public function foo(): self {
                         }
                     }',
             ],
+            'mixedTypeInferredIfErrors' => [
+                'code' => '<?php
+                    class A {}
+                    /**
+                     * @param A|string $a
+                     */
+                    function foo($a): void {
+                        /**
+                         * @psalm-suppress PossiblyInvalidClone
+                         */
+                        $cloned = clone $a;
+                    }',
```

Comment:
I think this test needs to return `$cloned`. and make sure MixedReturn is reported.